doc/nrf/protocols/matter/end_product/security.rst

```diff
@@ -7,14 +7,62 @@ Security
    :local:
    :depth: 3
 
-Nordic Matter samples leverage security features supported in the |NCS| that can be divided into three major categories:
+Nordic Matter samples leverage security features supported in the |NCS| that can be divided into four major categories:
 
+* Trusted Execution Environment
```

Comment:
There is nothing about "Trusted Execution Environment as a security feature" in the NCS based on https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/security.html . Should there be some information, @mia-ko ?

Reply:
Ok, there is no something like that. So how can I name it? I need a name for both the nRF54H20 secure domain and TF-M on nRF54L15. Both of them are isolated from the application firmware.

Reply:
Yes, that's why I poked @mia-ko and added @FrancescoSer and @annwoj to the PR so that we can see if this needs a mention on the Security page, mention in the device guides, or no mention at all (in which case this section can stay, but probably not with the current name or description, because it could be interpreted as TEE is a Matter-only feature in the NCS).

Reply:
Instead of using Trusted Execution Environment, why not align on using the terms that are already in use in our docs (SPE)?
Secure processing environment is a standard term in the PSA Security model.

Reply:
@greg-fer I will remove the nRF54H20 part for now. For this release, we need to describe TF-M only, and this commit it related to it.